### DISCUSSION TITLE
Use eslint_d as formatter in efm

### DISCUSSION THREAD

**Discussion by rafaelderolez** [CosmicNvim/CosmicNvim#8] [Ideas]
For the past few years, I've always used [eslint configs that contain the Prettier config](https://github.com/wesbos/eslint-config-wesbos/blob/master/.eslintrc.js#L2). This means I'm not using Prettier directly as a formatter.

As a test, I've added this (stolen from [here](https://www.reddit.com/r/neovim/comments/qckrnp/share_your_prettier_and_eslint_formatting_setup/hhhr8s1/)) to `lua/cosmic/lsp/providers/efm.lua`:
```lua
local eslint = {
  lintCommand = "eslint_d -f unix --stdin --stdin-filename ${INPUT}",
  lintStdin = true,
  lintFormats = {"%f:%l:%c: %m"},
  lintIgnoreExitCode = true,
  formatCommand = "eslint_d --stdin --fix-to-stdout --stdin-filename=${INPUT}",
  formatStdin = true
}
```

The above seems to work well, but it means that both options from `:lua vim.lsp.buf.formatting()` now end up doing the same thing. Would there be a way to 'turn off' the eslint one? Does it make sense to do so? I'd also like to set up formatting on save somewhere this week. Would be nice to find some common ground here and maybe bake it into Cosmic if you're interested.

![image](https://user-images.githubusercontent.com/6158959/138596961-8d6e2025-686a-4b32-a7f5-9e37e26c0165.png)

**Comment by mattleong** [comment]
@rafaelderolez I think this is a very reasonable idea. In fact, I was using efm for eslint formatting as well just a few weeks ago and I do find the need to choose between the two annoying as well. 

Since I imagine a lot of people are still using efm for both prettier and eslint, it might be prudent to add support for that. I'll drum up a couple of issues to track. 

But first the biggest question is how to expose that? Write documentation to change the code directly...? Or how would you feel about a simple config file where you could define some basic settings? Something like below:

```lua
-- config.lua
local config = require('cosmic.config')
config.lsp.format_on_save = true -- enable format on save
config.lsp.servers.eslint = false -- disable eslint lsp
config.lsp.servers.efm.format = {'eslint', 'prettier'} -- chose formatters (will use eslint_d and prettierd)
```

**Reply by mattleong** [reply]
Issue created: https://github.com/mattleong/CosmicNvim/issues/9

**Reply by rafaelderolez** [reply]
I think a config file/folder is the way to go. I looked into NvChad the other day and really liked how it has a `custom` folder you can use to do whatever, while keeping the ability to update itself.

**Reply by mattleong** [reply]
I like that too. I have a branch in progress now that will add the concept of a config file. Hopefully should have it fully stubbed out by tomorrow. 

Feel free to create issues for any other settings you'd find useful!

**Reply by mattleong** [reply]
@rafaelderolez new config folder can be found in `./lua/cosmic/config` :) 

I'm definitely liking the customizability that it provides. Let me know what you think!
